plugins/org.python.pydev/src/org/python/pydev/editor/codefolding/CodeFoldingSetter.java

```diff
@@ -292,6 +293,13 @@ public void propertyChanged(Object source, int propId) {
     private static final Pattern regionStartPattern = Pattern.compile("(\\s)*#(\\s)*\\bregion\\b");
     private static final Pattern regionEndPattern = Pattern.compile("(\\s)*#(\\s)*\\bendregion\\b");
 
+    private static boolean getBooleanPreference(String key) {
```

Comment:
I think you should pass `IAdaptable projectAdaptable` here... it seems trivial to pass it to `getMarks` to receive it here (the test can still call `getMarks` with null, but in the real-world use case the `CodeFoldingSetter` has an `editor` attribute.

Reply:
Aah, I did not see that member, will use that.